### ISSUE TITLE
Similar project

### ISSUE THREAD

**Issue by NiKiZe** [piedar/ipfs-distfiles-gentoo#1]
Hi,
Just want to let you know that I'm working on something similar, at least probably the same end goal.
See ipfs/notes#296
I'm currently working on getting a full IPFS mirror of the distfiles up and running at https://github.com/NiKiZe/Gentoo-distfiles-IPFS

I will close this issue directly after creating it, since it's not a real issue, just wanted to let you know.

**Comment by piedar** [comment]
Cool!  I think your import script is better because it syncs directly from a distfiles mirror instead of downloading each package's sources one by one.  I got bored running mine on all the packages...

I started on another project to [mount ipfs filesystems](https://github.com/piedar/js-ipfs-mount), which could be complementary once the [chunking situation](https://github.com/piedar/js-ipfs-mount/issues/6) is solved.  It would also be cool to define a way to download and verify files by hash directly in the ebuild, possibly removing the need for additional Manifest files.

**Comment by NiKiZe** [comment]
Regarding IPFS hash in ebuild, I think that is a bad place, for the same reason that other checksum are stored in Manifest instead of ebuild (the hash should be created by external tools from the data without needing manual modifications) So I'm voting for adding them to the Manifest, in the ipfs notes I mentioned you will find a link to the Gentoo forums where I have tried to explain this to some other people.
One middle ground is to generate full file to ipfs hash mappings which could be distributed separately. (which is on my todo list)

If ipfs mounts can be faster, than that's great! does official IPFS devs know about this project?
The chunking should "always" be done with 256K as it's the default so maybe you should run internal buffer for this? (I probably don't know what I'm talking about here but that is what I have understood so far)
This support writes as well, but read-only support should be good enough, and that should not have the chunking issues?
Is there IPNS support as well?

**Comment by piedar** [comment]
> Regarding IPFS hash in ebuild, I think that is a bad place, for the same reason that other checksum are stored in Manifest instead of ebuild (the hash should be created by external tools from the data without needing manual modifications)

My hope for that was to take full advantage of content addressing.  But reading that forum thread I see there is no way Gentoo devs will agree to start adding hashes to ebuilds in that fashion.  But that's OK - I've found other things to work on and I'm not so worried about the distfiles problem at the moment.

> The chunking should "always" be done with 256K as it's the default so maybe you should run internal buffer for this?

I tried that a couple ways in feature branches but it isn't stable yet.  Hopefully [upstream can solve it](https://github.com/ipfs/go-ipfs/issues/5129) instead.

> Is there IPNS support as well?

[Not yet](https://github.com/piedar/js-ipfs-mount/issues/1), but I think that will be the next big feature when I next get some time to work on it.
